### DISCUSSION TITLE
Navigation voice instructions

### DISCUSSION THREAD

**Discussion by myfta** [osmandapp/OsmAnd#20672] [Maps Related]
Where does OsmAnd get its street names from when giving guided voice navigation? 

I have a street that is incorrectly named. If I dive into OSM directly the street is named correctly. Is there maybe a text-to-speech file that has not been updated on my device? Or perhaps those sorts of files do not get updated very often, if so where are the master files held?

**Comment by vshcherb** [comment]
From name, name:en tags of OpenStreetMap.

**Reply by myfta** [reply]
That's what I would have expected. It is correctly named on OSM, but if you search the location on OsmAnd it gives an incorrect name. It feels like there is a look-up database that is not being updated with the addresses.

**Reply by tommycrock** [reply]
Perhaps give some details like what is it called and where is it. Have you updated your maps? When/if it changed in OSM.

**Reply by myfta** [reply]
This is not the only one I have found.

On OsmAnd if I do a search at this location https://osmand.net/map?pin=51.06878,-0.34016#18/51.06878/-0.34016 it returns an address of "Merryfield Way" when actually it is "Merryfield Drive".
This is the corresponding OSM location https://www.openstreetmap.org/?mlat=51.06878&mlon=-0.34016#map=18/51.06878/-0.34016 
On OSM it is named correctly and I cannot find an erroneously named feature.

What I have noticed on the OsmAnd map both on mobile and on the web there are a number of Points/Nodes that do not appear on OSM and these are certainly erroneously named.

I have found at least two other streets similarly incorrectly named in the area. Here is an example.
<img width="418" alt="Screenshot 2024-08-26 140348" src="https://github.com/user-attachments/assets/be8d6d13-878d-4d82-ad94-36ad6cc2bfa3">

**Reply by tommycrock** [reply]
I also see this.
![Screenshot_20240826_155430_OsmAnd+](https://github.com/user-attachments/assets/750a989a-1b18-4acb-b503-b60f6a18a4a7)

Merryfield Way is a service road also in Horsham but not very near Merryfield Drive https://www.openstreetmap.org/way/209549942

This sounds similar to the bug that was reported regarding a road in Germany where the name indexing is overly aggressive in grouping different spellings?

**Reply by myfta** [reply]
So do you think this is grouping rather than the index itself? It's like you search for one place and it sends you somewhere else. 

We definitely don't want places 26 km apart being grouped! However, if we make it too short there might be a few local issues where streets do have a small road off of them suffixed by 'close' or 'way'.

Unfortunately, there is no taxonomy for street naming in the UK, and probably elsewhere, too.

**Reply by tommycrock** [reply]
As I understand it the index is built from the map data. It looks like it might be the same problem to me but let's see what the devs think.

**Reply by tommycrock** [reply]
This was the issue I was thinking of #12602 but that's a much closer name

**Reply by myfta** [reply]
Is there an action to fix this any time soon?
